graphqlapi/src/main/java/io/stargate/graphql/schema/schemafirst/migration/MigrationQuery.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.schema.schemafirst.migration;
+
+import io.stargate.db.query.builder.AbstractBound;
+
+/** A DDL query to be executed as part of a migration. */
+public class MigrationQuery implements Comparable<MigrationQuery> {
+
+  private final AbstractBound<?> query;
+  private final String description;
+  private final boolean isAddColumn;
```

Comment:
we don't allow removing columns from the schema, right?

Reply:
No. We could, but so far I've avoided it. My rationale is that someone might reuse an existing schema that's shared with other non-GraphQL applications, and not map every column. It would be a shame if GraphQL accidentally dropped a bunch of columns that were actually needed by something else.

With that being said, the `DROP_AND_RECREATE*` strategies _will_ drop columns in such a scenario, but I think their names are a clear enough indication that they are destructive.

The strategies are not final yet, we can revisit them. I think there might always be corner cases that are better handled with manual CQL queries. If it's just cleaning up obsolete columns that should be fine.